qcodes/instrument_drivers/yokogawa/GS200.py

```diff
@@ -517,6 +518,7 @@ def _set_source_mode(self, mode: str) -> None:
 
         self.write("SOUR:FUNC {}".format(mode))
         # Update the measurement mode
+        self.source_mode.cache.set(mode)
```

Comment:
Sorry one last thing. we should put a comment explaining why this is needed. 
```
# we set the cache here since `_update_measurement_module` needs the current value which would otherwise only be set after this method exits
```